src/information_panel.rs

```diff
@@ -0,0 +1,459 @@
+                            self.forget_last_stored_image(ui);
+                        }
+                    } else if let Some(mut content) = panel_entry.content() {
+                        egui::ScrollArea::vertical()
+                            .max_height(100.0)
+                            .show(ui, |ui| {
+                                ui.add(egui::TextEdit::multiline(&mut content).code_editor());
+                            });
+                    } else {
+                        // if now preview is available, show icon
+                        ui.vertical_centered(|ui| {
+                            ui.label(egui::RichText::from(item.icon()).size(120.0));
+                        });
+                    }
+                }
+            } else {
+                // if now ext is available, show icon anyway
+                ui.vertical_centered(|ui| {
+                    ui.label(egui::RichText::from(item.icon()).size(120.0));
+                });
+            }
+        }
+    }
+
+    fn forget_last_stored_image(&mut self, ui: &Ui) {
+        if let Some(last_image) = self.stored_images.first() {
+            ui.ctx()
+                .forget_image(format!("file://{last_image}").as_str());
+        }
+        self.stored_images.shift_remove_index(0);
+    }
+
+    /// removes all loaded preview images from the egui-loaders to reduce memory usage.
+    pub fn forget_all_stored_images(&mut self, ui: &Ui) {
+        for image in &self.stored_images {
+            ui.ctx().forget_image(format!("file://{image}").as_str());
+        }
+        self.stored_images.clear();
+    }
+
+    fn load_meta_data(&mut self, item: &DirectoryEntry) {
+        let path_buf = item.to_path_buf();
+        if self.loaded_file_name != path_buf {
+            self.loaded_file_name.clone_from(&path_buf);
+            // clear previous meta data
+            self.other_meta_data = IndexMap::default();
+            if let Some(ext) = path_buf.extension() {
+                if let Some(ext_str) = ext.to_str() {
+                    if let Some(load_meta_data) = self.additional_meta_files.get_mut(ext_str) {
+                        // load metadata
+                        load_meta_data(&mut self.other_meta_data, &path_buf);
+                    }
+                }
+            }
+            let content = self.load_content(path_buf);
+            self.panel_entry = Some(InfoPanelEntry::new(item.clone()));
+            if let Some(panel_entry) = &mut self.panel_entry {
+                // load content
+                if panel_entry.content().is_none() {
+                    *panel_entry.content_mut() = content;
+                }
+            }
+        }
+    }
+
+    fn display_meta_data(&self, ui: &mut Ui, width: f32, item: &DirectoryEntry) {
+        egui::ScrollArea::vertical()
+            .id_salt("meta_data_scroll")
+            .show(ui, |ui| {
+                egui::Grid::new("meta_data")
```

Comment:
Is it possible to use the ID of the file dialog as the parent ID?

Reply:
how can I retrieve it from `FileDialog`?

Reply:
I think `file_dialog.config_mut().id` or something similar should work

Reply:
```rust
        let meta_id = file_dialog.config_mut().id.unwrap_or_else(||egui::Id::from("meta_data_scroll"));
```

Yeah, this is always None:
```rust
file_dialog.config_mut().id
```

Reply:
Ye, the id via config is only set if the default id is overwritten via the builder method. Therefore we need an access method for `FileDialog::window_id`, there the current id used is stored

Reply:
Done!